packages/lib/modules/lbp/steps/useSendMetadataStep.tsx

```diff
@@ -0,0 +1,86 @@
+import { TransactionStep } from '@repo/lib/modules/transactions/transaction-steps/lib'
+import { useState } from 'react'
+import { VStack, Button } from '@chakra-ui/react'
+import { LabelWithIcon } from '@repo/lib/shared/components/btns/button-group/LabelWithIcon'
+import { useLbpForm } from '@repo/lib/modules/lbp/LbpFormProvider'
+import { useMutation } from '@apollo/client'
+import { CreateLbpDocument } from '@repo/lib/shared/services/api/generated/graphql'
+import { useLocalStorage } from 'usehooks-ts'
+import { LS_KEYS } from '@repo/lib/modules/local-storage/local-storage.constants'
+
+export function useSendMetadataStep(): TransactionStep {
+  const [, setIsStepActivated] = useState(false)
+  const [createLbp] = useMutation(CreateLbpDocument)
+  const [poolAddress] = useLocalStorage<`0x${string}` | undefined>(
+    LS_KEYS.LbpConfig.Address,
+    undefined
+  )
+  const [isComplete, setIsComplete] = useLocalStorage<boolean>(
```

Comment:
Just as a _far fetched_ suggestion, we can create a state machine to hide all the localstorage complexity.